source/compiler/sc1.c

```diff
@@ -3400,7 +3454,7 @@ SC_FUNC void check_tagmismatch(int formaltag,int actualtag,int allowcoerce,int e
 {
   if (!matchtag(formaltag,actualtag,allowcoerce)) {
     constvalue *tagsym;
-    char formal_tagname[sNAMEMAX+3]="none (\"_\"),",actual_tagname[sNAMEMAX+2]="none (\"_\")"; /* two extra characters for quotes */  
+    char formal_tagname[sNAMEMAX+4]="none (\"_\"),",actual_tagname[sNAMEMAX+3]="none (\"_\")"; /* two extra characters for quotes */  
```

Comment:
Assume that `sNAMEMAX` is `32` (which it is in our compiler). An identifier can have at most 31 characters. The 32nd character will be used as a null terminator for the longest identifier.

With 31 significant characters of the tagname, two quotes and a comma (for `formal_tagname`), the maximum length turns out to be 31(identifier) + 2(quotes) + 1 (comma) + 1 (null terminator) = 35.

`sNAMEMAX + 3` = 35

For `actual_tagname`, it'll be one lesser as there is no need for comma.

I can't see why the length has been increased by one for both.